### PR TITLE
Update to Java 11 & support JPMS

### PR DESCRIPTION
**Pull request by Mechite**
# Motivation
JPMS is used by the vast majority of publically released libraries. The performance benefits it provides over the generic classpath, and better compile-time verification is very important. Without native support for JPMS, nobody can release a public library using JPMS and Minio safely.

# Downsides
This will break support for Java 1.8, and Java 9 or over will be needed to use Minio. In the PR that I have created, Java 11 is the base version as it is the next LTS release after 1.8. Most people consider Java 1.8 as "ancient" by modern standards, but it is still used today in a lot of commercial software. The best way to maintain support will be to simply refer people to the older versions of the client, or the official S3 clients.

A hacky way people tend to retain support for Java 1.8 is to compile the module-info with Java 9/11, and the rest of the project with Java 1,8. This is not the most elegant build toolchain but is a valid solution nevertheless.

# Issues
This PR should not build correctly, as I have not thoroughly checked if all `requires` statements are defined, and there are likely several other issues.
This PR also uses `open module` for all of the declarations, as I was conducting some quick tests with Spring - this can be changed, and probably should be changed as the client does not seem to require any kind of reflection.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -108,8 +108,8 @@ subprojects {
 
     check.dependsOn localeTest
 
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
```
Comment (balamurugana) [inline build.gradle:112]: Our source code and minimum supported version is `1.8`.

File: adminapi/src/main/java/module-info.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```
Comment (balamurugana) [inline adminapi/src/main/java/module-info.java:2]: Year is `2023`

File: api/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+ * MinIO Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2020 MinIO, Inc.
```
Comment (balamurugana) [inline api/src/main/java/module-info.java:2]: Year is `2023`

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (Mechite) [reply build.gradle:112]: Aware.

**Comment by balamurugana** [comment]
Closing as stale PR

**Pull request closed without merging**